src/moonlink_service/moonlink-api.yaml

```diff
@@ -0,0 +1,1016 @@
+    #  ERROR RESPONSES
+    # ============================================================================
+    
+    BadRequest:
+      description: Bad Request - Invalid or malformed request
+      content:
+        application/json:
+          schema: { $ref: '#/components/schemas/ErrorResponse' }
+    
+    Unauthorized:
+      description: Unauthorized - Authentication required
+      content:
+        application/json:
+          schema: { $ref: '#/components/schemas/ErrorResponse' }
+    
+    Forbidden:
+      description: Forbidden - Insufficient permissions
+      content:
+        application/json:
+          schema: { $ref: '#/components/schemas/ErrorResponse' }
+    
+    NotFound:
+      description: Not Found - Resource not found
+      content:
+        application/json:
+          schema: { $ref: '#/components/schemas/ErrorResponse' }
+    
+    Conflict:
+      description: Conflict - Resource already exists
+      content:
+        application/json:
+          schema: { $ref: '#/components/schemas/ErrorResponse' }
+    
+    ServerError:
+      description: Server Error - Internal server error
+      content:
+        application/json:
+          schema: { $ref: '#/components/schemas/ErrorResponse' }
+
+  schemas:
+    # ============================================================================
+    # SCHEMAS
+    # ============================================================================
+    StorageRequest:
+      type: object
+      properties:
+        name: { type: string }
+        type: { type: string, enum: [s3, s3_compatible, gcs, filesystem] }
+        bucket: { type: string }
+        region: { type: string } # required for s3, optional for s3_compatible
+        endpoint: { type: string } # required for s3_compatible
+        default_path: { type: string, default: "" }
+        credentials:
+          type: object
+          properties:
+            access_key_id: { type: string }
+            secret_access_key: { type: string }
+            session_token: { type: string }
+          required: [access_key_id, secret_access_key]
+      required: [name, type, bucket, credentials]
+
+    Storage:
+      type: object
+      properties:
+        id: { type: string, description: "Unique identifier assigned by Moonlink for this storage configuration" }
+        name: { type: string, description: "User-friendly name for this storage configuration" }
+        type: { type: string, description: "Storage provider type" }
+        bucket: { type: string, description: "Name of the storage bucket/container" }
+        region: { type: string, description: "Storage region" }
+        endpoint: { type: string, description: "Custom endpoint URL (for S3-compatible storage)" }
+        default_path: { type: string, description: "Default path prefix within the bucket" }
+        configured_at: { type: string, format: date-time, description: "When this storage configuration was added to Moonlink" }
```

Comment:
From the field naming, it's hard to tell whether it's registration timestamp or update timestamp?
Also do you think it necessary to add a update time?